tests/meta-mender-ci/recipes-testing/mender-coverage-client/mender-client_%.bbappend

```diff
@@ -0,0 +1,67 @@
+SUMMARY = "Build a Mender client which records coverage during its invocation"
+
+DEPENDS += "gobinarycoverage rsync-native"
+
+do_copy_src_to_build[nostamp] = "1"
+do_copy_src_to_build[doc] = "Copies the Mender client source from ${S}/src/${GO_IMPORT} \
+to ${B}/src/${GO_IMPORT}"
+do_copy_src_to_build[dirs] = "${B}src/github.com/mendersoftware/mender"
+do_copy_src_to_build () {
+    bbwarn "Copying source to the build dir..."
+    rsync -a ${S}/src/${GO_IMPORT}/ ${B}/src/${GO_IMPORT}/
+    cd ${B}/src/${GO_IMPORT}/
+    # Why is this needed?
+    git reset --hard HEAD
+}
+
+do_instrument_client[dirs] =+ "${GOTMPDIR}"
+do_instrument_client[doc] = "Modifies the Mender client source to enable coverage analysis"
+do_instrument_client () {
+    # Since the Makefile installs its dependent tools using 'go get', add
+    # ${B}/bin to the path during the compilation
+    PATH="${B}/bin:$PATH"
+    export PATH
+    cd ${B}/src/${GO_IMPORT}
+    bbwarn "Building the Mender client with coverage analysis"
+    oe_runmake instrument-binary
+}
+
+python () {
+    # Only add coverage analysis if the client is newer than 2.2.x
+    if not mender_is_2_2_or_older(d):
+        bb.warn("Building the client with coverage instrumentation enabled")
+        # Disable do_configure inherited from go.bbclass. This function tries to
+        # symlink to the builddir from source i.e., ln -s ${S}/src ${B}/src.
+        d.setVarFlag('do_configure', 'noexec', "1")
```

Comment:
I would perhaps argue that it's better to be surgical and remove only the part we don't want, and not the whole function. We don't know what it does, or will do in the future, besides making the symlink. Can you simply use `do_configure_append` and do the link removal and copying there?

Reply:
Smrt!
Way better than this approach. Lets do create the inverse function!